(RE-13292) Add script to create and push enterprise-dist rc tags

sarameisburger opened the pull request.

sarameisburger commented:
I think this needs to be merged in for me to test it in the job (https://cinext-jenkinsmaster-pipeline-prod-1.delivery.puppetlabs.net/job/enterprise-dist_release_branch_creation/) is that true @cthorn42 ?

sarameisburger commented:
Or am I able to test it from my own fork

cthorn42 commented on vars/bash/tag_enterprise_dist_next_rc.sh:
```diff
@@ -0,0 +1,27 @@
+branch_from=$2
+
+rm -rf ./${GITHUB_PROJECT}
+git clone git@github.com:puppetlabs/${GITHUB_PROJECT} ./${GITHUB_PROJECT}
```
I think you're missing setting the GITHUB_PROJECT, unless somehow that is being passed in as a env variable?

cthorn42 replied:
Oh it is. For clarity might be easier just to declare that in this file. Either way should obviously work.

cthorn42 commented:
It kinda depends on what you're looking on testing out. 

I'm going to be on vacation all next week so I'm going to enter way to much text here. Feel free to ask me tomorrow if you get around to looking at this, or if I'm gone asking anyone from the Installer team question.

The way our jenkins pipeline instance, https://cinext-jenkinsmaster-pipeline-prod-1.delivery.puppetlabs.net/, is setup is it is configured to look at the puppetlabs github project. 
This means if you want to test out your enterprise-dist PR, push your branch sarameisburger:release-branch/2018.1.x/create to puppetlabs:release-branch/2018.1.x/create.
Since it is just another branch, hopefully enterprise-dist is safe to test out on? 
(i'm assuming this is your first pipelines job you're creating, if it isn't I apologize and skip the next few lines).
Anyway then you need to create a Jenkins Job that will be configured to use the release-branch branch of enterprise dist. So goto the jenkins instance above and click on 'New Item', enter the name of the job you're creating a the top, then scroll down to the bottom to Copy From. 
I'd recommend copying from because it will use all the settings for a known job, like `release_branch_creation_kickoff`.
From there you need to configure a few things under the Pipeline section.
* Repository URL, needs to be updated to point to your enterprise dist
* Branches to build,  needs to point to the branch you created
* Script path, the shared libraries script your adding as part of this PR.

If you want to test the script in this PR create a branch off of this repo like I described above for enterprise dist. Then update the groovy file as done in this PR, https://github.com/puppetlabs/pe-installer-shim/commit/6814cc74779a90353c67c41b2ebfa1a6647e24f7#diff-27a06a9e3d5e7f67eb604a39536208c9, so your groovy script will known to use the branch.

Then you should be able to just kick off the jenkins pipeline job and it see how it shakes out.

sarameisburger commented:
```
+ echo 'checking out release branch 9999.9.9-release'
checking out release branch 9999.9.9-release
+ git checkout 9999.9.9-release
Switched to a new branch '9999.9.9-release'
Branch 9999.9.9-release set up to track remote branch 9999.9.9-release from origin.
+ echo 'calling rake commands'
calling rake commands
+ bundle exec rake new_release:push_empty_commit PE_BRANCH_NAME=9999.9.9-release
Creating and pushing empty commit to 9999.9.9-release
+ bundle exec rake new_release:create_and_push_rc_tag PE_BRANCH_NAME=9999.9.9-release
From github.com:puppetlabs/enterprise-dist
 * branch                9999.9.9-release -> FETCH_HEAD
Tagging with '9999.9.9-rc19'
+ echo 'checking out mainline branch 9999.9.x'
checking out mainline branch 9999.9.x
+ git checkout 9999.9.x
Switched to a new branch '9999.9.x'
Branch 9999.9.x set up to track remote branch 9999.9.x from origin.
+ echo 'calling rake task...'
calling rake task...
+ bundle exec rake new_release:push_empty_commit PE_BRANCH_NAME=9999.9.x
Creating and pushing empty commit to 9999.9.x
+ echo 'deciding if this is an x or y release'
deciding if this is an x or y release
+ [[ 9999.9.x == \m\a\s\t\e\r ]]
+ bundle exec rake new_release:create_and_push_new_z_tag PE_BRANCH_NAME=9999.9.x
From github.com:puppetlabs/enterprise-dist
 * branch                9999.9.x   -> FETCH_HEAD
Tagging with '9999.9.10-rc0'
```

sarameisburger commented:
Tested this with https://cinext-jenkinsmaster-pipeline-prod-1.delivery.puppetlabs.net/job/enterprise-dist_release_branch_creation/
I commented out the actual parts that tag and push things (which have been used and tested before) and made sure the information being passed was correct. 
This pairs with these changes: https://github.com/puppetlabs/enterprise-dist/pull/3666/files that i'll eventually get merged into 2018.1.x and master

sarameisburger commented:
Ha whoops these comments are supposed to go on another PR :)
